pyop2/host.py

```diff
@@ -234,17 +310,17 @@ def c_vec_init(self, is_top, layers, is_facet=False):
                         vec_idx += 1
         return ";\n".join(val)
 
-    def c_addto(self, i, j, buf_name, tmp_name, tmp_decl,
-                extruded=None, is_facet=False, applied_blas=False):
+    def ast_addto(self, i, j, buf_name, tmp_name, tmp_decl,
+                  extruded=None, is_facet=False, applied_blas=False):
         maps = as_tuple(self.map, Map)
         nrows = maps[0].split[i].arity
         ncols = maps[1].split[j].arity
-        rows_str = "%s + i * %s" % (self.c_map_name(0, i), nrows)
-        cols_str = "%s + i * %s" % (self.c_map_name(1, j), ncols)
+        rows_ast = ast.Sum(ast.Symbol(self.c_map_name(0, i)), ast.Prod(ast.Symbol("i"), ast.Symbol(nrows)))
```

Comment:
Literals are mapped to symbols implicitly, and this works well if you don't have ranks or offsets.
So for example ast.Prod(ast.Symbol("i"), ast.Symbol(nrows)) == ast.Prod('i', nrows)